packages/hash/frontend/src/pages/[account-slug]/types/entity-type/[entity-type-id]/entities.page.tsx

```diff
@@ -0,0 +1,167 @@
+  const { entities, entityTypes, propertyTypes, subgraph } =
+    useEntityTypeEntities() ?? {};
+
+  const [showSearch, setShowSearch] = useState(false);
+  const [tableSort, setTableSort] = useState<TableSort<string>>({
+    key: "entity",
+    dir: "desc",
+  });
+
+  const namespace = useRouteNamespace();
+
+  const { columns, rows } =
+    useEntitiesTable(entities, entityTypes, propertyTypes, subgraph) ?? {};
+
+  const entitiesCount = useMemo(() => {
+    const namespaceEntities =
+      entities?.filter((entity) => entity.ownedById === namespace?.id) ?? [];
+
+    return {
+      namespace: namespaceEntities.length,
+      public: (entities?.length ?? 0) - namespaceEntities.length,
+    };
+  }, [entities, namespace]);
+
+  const drawHeader = useDrawHeader(tableSort, columns ?? []);
+
+  if (!columns || !rows) {
+    return null;
+  }
+
+  const handleHeaderClicked = createHandleHeaderClicked(
+    columns,
+    tableSort,
+    setTableSort,
+  );
+
+  const sortedRows = rows && sortRowData(rows, tableSort);
+
+  return (
+    <Box>
+      <SectionWrapper
+        title="Entities"
+        titleTooltip="This table lists all entities with the ‘Company’ type that are accessible to you"
+        titleStartContent={
+          <Stack direction="row">
+            {entitiesCount.namespace || entitiesCount.public ? (
+              <Stack direction="row" spacing={1.5} mr={2}>
+                {entitiesCount.namespace ? (
+                  <Chip
+                    size="xs"
+                    label={`${entitiesCount.namespace} in @${namespace?.shortname}`}
+                    icon={<HomeIcon />}
+                    sx={({ palette }) => ({ color: palette.gray[70] })}
+                  />
+                ) : null}
+
+                {entitiesCount.public ? (
+                  <WhiteChip
+                    size="xs"
+                    label={`${entitiesCount.public} public`}
+                    icon={<EarthIcon />}
+                  />
+                ) : null}
+              </Stack>
+            ) : null}
+
+            <Stack direction="row" spacing={0.5}>
+              <IconButton
+                rounded
+                onClick={() => setShowSearch(true)}
+                sx={{ color: ({ palette }) => palette.gray[60] }}
+              >
+                <FontAwesomeIcon icon={faMagnifyingGlass} />
+              </IconButton>
+            </Stack>
+          </Stack>
+        }
+        tooltipIcon={
+          <FontAwesomeIcon icon={faCircleQuestion} sx={{ fontSize: 14 }} />
+        }
+      >
+        <Paper sx={{ overflow: "hidden" }}>
+          <GlideGrid
+            showSearch={showSearch}
+            onSearchClose={() => setShowSearch(false)}
+            onHeaderClicked={handleHeaderClicked}
+            drawHeader={drawHeader}
+            columns={columns}
+            rows={sortedRows.length}
+            getCellContent={([colIndex, rowIndex]) => {
+              if (sortedRows && columns) {
```

Comment:
Two things here

- Having this function defined above at the component body instead JSX makes it easier to read imo. Let me know what do you think.
- I don't think we need this `if (sortedRows && columns)` check here, since it's already checked above where we `return null` if these are falsy.